rfc-176-replacing-static.md

```diff
@@ -0,0 +1,127 @@
+---
+status: proposed
+implementation: proposed
+status_last_reviewed:
+---
+
+# Replacing Static
+
+## Summary
+
+We propose two new gems: `govuk_web_banners` and `govuk_web_layout` which will replace the [slimmer] gem, the [static] application, and will give us the option if necessary to remove parts of the [govuk_publishing_components] gem that might not belong there. Between them they will handle static’s four main responsibilities: banners, static assets, layouts, and headers/footers, but in a less astonishing way.
```

Comment:
(random aside: today I learned you can't type repsonses on github if you've turned off javascript)

I want to disagree with this. I think that we should be more specific in what our gems are called when they are directly addressing the web problem. Keith's original name for this clashed with govuk_frontend which is the boggling name for the Design System code, and if we went with your suggestion then it's entirely unclear what the scope of the gem is and it could clash with future products - if we decided to spin up a forms or notification gem we'd hit such problems with your suggestion. This is probably why we have govuk_publishing_components but publishing is a loaded term within the GOV.UK programme, and this is specifically addressing a web need divorced from the publishing tools.

Reply:
`govuk_publishing_banners` and `govuk_publishing_layout`?

Reply:
I can sort of see your point but `govuk_web` isn't exactly our name and none of our other gems have that - so there'd be nothing to indicate that it was connected to us rather than forms or notify, nor using an existing precedent so just seems like a quirk of inconsistency. I'd be more inclined to agree if we were establishing a convention to apply to all our gems.

I do recall the naming of govuk-frontend was very decisive when design system came up with it and people in this programme tried to stop it because of it clashing with our conventions

Reply:
I think I would quite like to establish that naming, actually! Particularly as web is being considered a platform alongside mobile, and this work is specific to the website and not the publishing tools side of things within the programme.

Reply:
I realised late yesterday that I had been making this point while viewing the layouts as being solely website layouts because they're the ones using static. This does raise the question of what happens to the admin/publishing layouts which Kevin asks about further on. Options seem to be 1) leave them where they are. Seems a bit broken if we're deciding layouts don't belong among other components, 2) move them with the web layouts, use a name that reflects use in both places, 3) have a twin gem specifically for publishing - seems like too much duplication, 4) consider whether publishing apps might directly implement their own layout - depends whether there's enough shared between them to support it being elsewhere